ember.md

```diff
@@ -0,0 +1,490 @@
+
+### All functions must be named
+
+Any function declared with the `function` keyword must have a name. Functions declared with the es6 short "fat arrow" syntax do not need a name.
+
+This helps with debugging, as the name of the function shows up in the backtrace instead of `(anonymous function)`.
+
+When the function is used as part of a macro to define a property, we have a convention of using the property name prefixed with an underscore.
+
+```js
+// Good
+
+things.map((thing) => get(thing, 'name'));
+
+fullName: computed('firstName', 'lastName', function _fullName() {
+  // Code
+}),
+
+// Bad
+
+fullName: computed('firstName', 'lastName', function() {
+  // Code
+}),
+```
+
+## Organizing your modules
+
+### ES6 module files
+
+1. __Framework imports__
+
+   First, import Ember and ember addons.
+
+2. __Internal imports__
+
+   Then, separated by a blank line, import internals.
+
+3. __Destructure constants__
+
+4. __Object definition__
+
+5. __Utility functions__
+
+   After the object definition, at the bottom of the file, place any private utility functions.
+
+```
+import Ember from 'ember';
+
+import x from 'fern/utils/x';
+
+const { computed } = Ember;
+
+export default Ember.Object.extend({
+  // all the properties, this is where the list we're talking about goes
+});
+
+function _someUtilityFunction() {
+}
+```
+
+### Ember Object Properties
+
+Ordering a module's properties in a predictable manner will make it easier to
+scan.
+
+1. __Plain properties__
+
+   Start with properties that configure the module's behavior. Examples are
+   `tagName` and `classNames` on components and `queryParams` on controllers and
+   routes. Followed by any other simple properties, like default values for properties.
+
+2. __Single line computed property macros__
+
+   E.g. `alias`, `sort` and other macros. Start with service injections. If the
+   module is a model, then `attr` properties should be first, followed by
+   `belongsTo` and `hasMany`.
+
+3. __Multi line computed property functions__
+
+4. __Lifecycle hooks__
+
+   The hooks should be chronologically ordered by the order they are invoked in.
+
+5. __Functions__
+
+   Public functions first, internal functions after.
+
+6. __Actions__
+
+```js
+export default DS.Component.extend({
+  // Plain properties
+  tagName: 'span',
+
+  // Single line CP
+  post: alias('myPost'),
+
+  // Multi line CP
+  authorName: computed('author.{firstName,lastName}', function _authorName() {
+    // Code
+  }),
+
+  // Lifecycle hooks
+  didReceiveAttrs() {
+    this._super(...arguments);
+    // Code
+  },
+
+  // Functions
+  someFunction() {
+    // Code
+  },
+
+  _privateInternalFunction() {
+    // Code
+  },
+
+  actions: {
+    someAction() {
+      // Code
+    },
+  },
+});
+
+// Utility functions should go after the class
+
+function _someUtilityFunction() {
+  // Code
+}
+```
```

Comment:
This organization looks great to me -- this formalizes what we've (mostly) been doing